salt/modules/dnsutil.py

```diff
@@ -245,17 +245,42 @@ def A(host, nameserver=None):
 
     .. code-block:: bash
 
-        salt ns1 dig.A www.google.com
+        salt ns1 dnsutil.A www.google.com
     '''
     if _has_dig():
         return __salt__['dig.A'](host, nameserver)
     elif nameserver is None:
         # fall back to the socket interface, if we don't care who resolves
         try:
-            (hostname, aliases, addresses) = socket.gethostbyname_ex(host)
+            addresses = [sock[4][0] for sock in socket.getaddrinfo(host, None, socket.AF_INET, 0, socket.SOCK_RAW)]
             return addresses
-        except socket.error:
-            return 'Unabled to resolve {0}'.format(host)
+        except socket.gaierror:
+            return 'Unable to resolve {0}'.format(host)
+
+    return 'This function requires dig, which is not currently available'
+
+
+def AAAA(host, nameserver=None):
+    '''
+    Return the AAAA record(s) for `host`.
+
+    Always returns a list.
+
```

Comment:
There you go :+1: 

We'll see; If deemed more appropriate, I'll PR against 2015.2.